### ISSUE TITLE
FR: Add option to change the prefix of branch names generated by `jj git push --change`

### ISSUE THREAD

**Issue by tp-woven** [martinvonz/jj#340]
In our repo, we have a ("soft") convention where branch names are prefixed with the user name to make it easy to tell who "owns" which branch. It would be good if there was a config setting to override the `push-` prefix in the generated branch names.

(I'm happy to try and add this myself, if you think this would be a useful feature 😄)

**Comment by martinvonz** [comment]
> (I'm happy to try and add this myself, if you think this would be a useful feature 😄)

Yes, that sounds useful! (Both the feature and you doing it 😄  Thanks!)

**Comment by martinvonz** [comment]
If you have any questions about implementing it, it may be easiest to ask on https://discord.gg/dkmfj3aGQN, but also feel free to ask here.

For now, I'll just note that since libgit2 doesn't support running a local server to push to, and we don't use the `git` binary for that either, feel free to just test manually.
